include/torch-mlir/Dialect/Torch/IR/TorchOps.td

```diff
@@ -897,8 +899,9 @@ def Torch_OverwriteTensorOp : Torch_Op<"overwrite.tensor", [
     might mutate `overwritten`, so this relationship need not hold for the
     entire program.
 
-    This op has undefined behavior if the two tensors have different
-    shapes or dtypes.
+    This op does not have the AllowsTypeRefinement trait because the types of the
+    two operands are coupled. Only places that know how to simultaneously update
```

Comment:
You might want to also mention / tighten the wording in this ODS description that this op only updates the tensor data (not metadata), and thus cannot change the (dynamic) shape of the overwritten tensor. There are cases in PyTorch that require modeling in-place updates to the metadata/shape, but we want to clearly isolate those with a new op when we start to model them (maybe we should rename this op torch.overwrite.tensor.contents).

This avoids complexities for a case like:

```
%0 : !torch.vtensor<[?],f32>
%1 : !torch.tensor<[2],f32>
torch.overwrite.tensor %0 overwrites %1
```

We want torch.overwrite.tensor.contents to have undefined behavior if %0  is not the same dynamic shape/dtype as %1 (which in this case means that %0 can be assumed to be [2],f32 at the program point where torch.overwrite.tensor executes). It's very similar to tensor_static_info_cast'ing `[?]` to `[2]` in terms of what we are allowed to assume.  torch.overwrite.tensor.contents is inserted internally by the compiler in a very small number of places, so maintaining these strong invariants isn't too big of a deal, I hope.

Reply:
Thanks for the explanation! I've updated the ODS description and renamed the op to `torch.overwrite.tensor.contents`.